### PR TITLE
Update to the contributing.md file

### PR DESCRIPTION
**Pull request by Chizobaonorh**
I made significant changes and rewrote the contributing.md file for easy comprehension, and also added graphic images for easy navigation. @gbowne1  @LOGESH-B  @BlackBond06

### CODE REVIEW

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:12]: There are no strict requirements for documentation skills, good spelling, punctuation and grammer or using a spelling and grammar checker tool are all that are required, other than knowlege of  the  Markdown format.

**Review by BlackBond06 — COMMENTED** [review]
I would also suggest you use easy and simple to understand words.

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:28]: This could be easier to write if we change `If you notice a typo, slight bug in the code and there are no lodged complaints on this yet, select the issues icon at the top, and on the right side of the next page select new issues and lodge your complain.` 

to:

`Have you noticed something that did not look or work right in our applicaton, or a typographical error, wrong syntax, grammar, spelling or other issue?  

- Check to see if an issue has been created for whatever you have found  in <https://github.com/gbowne1/codebooker/issues>
- If there is not a issue that has already been created, Click on the green [New Issue] button.
- Describe your issue as well as you can:
  - include code snippits where the issue
  - include screen shots
  - include a short video
  
  If the issue appears to be a browser, operating system, device specific issue, let us know what those are.

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:34]: This might be a great place in the future to provide instructions for both cloning and forking.  They are similar process, but work differently especially when it comes to version control.  "Fork" was meant to be used for a derivitive project, and also meant for developers who wished to make their own version of a project, or a developer could use parts of a project to create their own project within the terms of the original project's license of course..   "Clone" is just that, a `clone` of the project, not meant to be used for creating any derivitive project that would be different from the main repository.

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:54]: DEVSETUP.md will be a file used for:

Instructions for developers, programmers, coders to:

Clone or Fork the project
Set up the project in their IDE or editor
Set up their environment to work in the project, including any environment variables, yarn/npm/pnpm package(s), editor extensions or plugins needed
Setting up the connection to the database
Setting up the local development database, MongoDB

Basically any specific instructions a beginner or even an advanced developer/programmer needs to know to get started working in the project.

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:56]: This section doesn't appear to be using text for using the database.

The original heading I had ### PR's was intended to create short instructions on how to create and submit Pull Requests (PR's), this appears now at line 70

File: CONTRIBUTING.md
Comment (gbowne1) [inline CONTRIBUTING.md:67]: This section needs to be removed and included in the section about creating an issue.

**Review by gbowne1 — CHANGES_REQUESTED** [review]
- [x] Checked out this documentation Pull Request change locally for review

I have made some comments about the text for you to review, comment back if you wish for clarification, etc.

I am requesting some minor changes here.  However, the majority of this change looks good.  The rest is fine.

You can commit changes to your PR branch to update and I will re-review..  Thank you for the opportunity to review your PR @Chizobaonorh!  

We are currently requiring a minumim of two PR approvals to merge from @LOGESH-B and @BlackBond06.  

If you need anything else, use the Slack documentation channel.

File: CONTRIBUTING.md
Comment (Chizobaonorh) [reply CONTRIBUTING.md:28]: Okay this is a much better alternative, will get this fixed.

File: CONTRIBUTING.md
Comment (Chizobaonorh) [reply CONTRIBUTING.md:34]: I felt this would make this page more choked up, i wanted to provide information to this on the devsetup page, or what do you think?

File: CONTRIBUTING.md
Comment (Chizobaonorh) [reply CONTRIBUTING.md:54]: Exactly, this was my exact same thought process, that is why i did not expantiate on this in this particular file.

File: CONTRIBUTING.md
Comment (Chizobaonorh) [reply CONTRIBUTING.md:56]: I felt placing it after instructions on forking this repository and setting up the IDE was ideal, since they follow suit

File: CONTRIBUTING.md
Comment (Chizobaonorh) [reply CONTRIBUTING.md:67]: Okay i will do that.

**Comment by Chizobaonorh** [comment]
> @gbowne1 @Chizobaonorh
> 
> I can't find any steps to setup the .env for backend and instruction to run the setup.js file. It is very important to setup this project locally

I am not sure this was meant to be included in the contributing file, my best guess would be for this to be included in the devsetup file

**Comment by gbowne1** [comment]
@LOGESH-B This file CONTRIBUTING.md is basically all about how to contribute which would include only information on how to clone/fork, pull requests, making bug reports, creating issues, maybe even links to the TODO list

We should include somehow within the file, that we require a minimum of two (2) reviews to be completed before we will merge.

DEVSETUP.md can house all of the things necessary for the developer to get their environment  set up and working.

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:34]: Yes, @Chizobaonorh, that information for would actually be better for the DEVSETUP.md file

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:67]: @Chizobaonorh Ok.

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:56]: @Chizobaonorh I agree.

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:54]: @Chizobaonorh Good. Hopefully we can get to the DEVSETUP.md file soon.

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:28]: @Chizobaonorh Ok. Let me know when you have updates.

**Review by gbowne1 — APPROVED** [review]
The changes to this PR look good.  

Some things ideally could be summarized, though.

Approving for merge.

**Comment by Chizobaonorh** [comment]
> The changes to this PR look good.
> 
> Some things ideally could be summarized, though.
> 
> Approving for merge.

Can you point the things you think i can summarize?

**Comment by gbowne1** [comment]
Lets wait for that till we finish what we have.

File: CONTRIBUTING.md
Comment (LOGESH-B) [inline CONTRIBUTING.md:18]: I can't able to understand on 1st read

**Review by LOGESH-B — APPROVED** [review]
Hii @Chizobaonorh 
Thanks for your valuable time to complete this docs
Approving for merge🤝

File: CONTRIBUTING.md
Comment (gbowne1) [reply CONTRIBUTING.md:18]: Yeah, that wording is confusing, and could probably be removed.

Comment (Chizobaonorh) [reply CONTRIBUTING.md:18]: > I can't able to understand on 1st read

Okay will work on this

Comment (Chizobaonorh) [reply CONTRIBUTING.md]: > Yeah, that wording is confusing, and could probably be removed.

Yeah i have simplified this, it reads better now.

Comment (gbowne1) [reply CONTRIBUTING.md]: I just saw. Looks much better.  The original text was placed there because I had so many people in this (and other projects too) ask me if they could please work on a issue.  I understand that is just being polite and that was fine, but it is not necessary that people ask for permission to work on an issue.  I largely prefer that people just say they would like to work on an issue, and one of our maintaners can assign.

Comment (Chizobaonorh) [reply CONTRIBUTING.md]: Yes, i totally agree with that

**Review by BlackBond06 — APPROVED** [review]
Updates made in this PR greatly improves the contribution.md file, enhancing readability and comprehension. Thanks for the opportunity to review you PR @Chizobaonorh. Approving for merge.

**Comment by gbowne1** [comment]
Alright, merging now. Thanks for your contribution @Chizobaonorh and the opportunity to review your PR.

**Pull request merged**
